### PR TITLE
Initial project

### PR DESCRIPTION
**Pull request by ndt0409**
## Related Tickets
- [#Task 55571: Initial project](https://edu-redmine.sun-asterisk.vn/issues/55571)
## WHAT
- apply mvvm
- splash screen
- bottom navigation 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/values/colors.xml [hunk 1]
```diff
@@ -5,6 +5,8 @@
     <color name="purple_700">#FF3700B3</color>
     <color name="teal_200">#FF03DAC5</color>
     <color name="teal_700">#FF018786</color>
-    <color name="black">#FF000000</color>
-    <color name="white">#FFFFFFFF</color>
+    <color name="color_black">#FF000000</color>
+    <color name="color_white">#FFFFFFFF</color>
+
+    <color name="color_red">#C01E21</color>
 </resources>
```
Comment (hungld-1302) [inline app/src/main/res/values/colors.xml]: endline

File: app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="dp_0">0dp</dimen>
+</resources>
```
Comment (hungld-1302) [inline app/src/main/res/values/dimens.xml]: .

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1,3 +1,7 @@
 <resources>
     <string name="app_name">Marvel App</string>
+    <string name="title_home">Home</string>
+    <string name="title_comic">Comic</string>
+    <string name="title_favorite">Favorite</string>
+    <string name="title_setting">Setting</string>
```
Comment (hungld-1302) [inline app/src/main/res/values/strings.xml:6]: .

File: app/src/main/res/values/themes.xml [hunk 4]
```diff
@@ -1,16 +1,21 @@
...
         <!-- Status bar color. -->
         <item name="android:statusBarColor">?attr/colorPrimaryVariant</item>
         <!-- Customize your theme here. -->
     </style>
+
+    <style name="SplashTheme" parent="Theme.MarvelApp">
+        <item name="android:windowFullscreen">true</item>
+        <item name="android:windowBackground">@drawable/bg_splash</item>
+    </style>
 </resources>
```
Comment (hungld-1302) [inline app/src/main/res/values/themes.xml]: .

File: app/src/main/res/layout/fragment_home.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".ui.HomeFragment">
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (hungld-1302) [inline app/src/main/res/layout/fragment_home.xml]: .

File: app/src/main/res/menu/menu_bottom_navigation.xml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+    <item
+        android:id="@+id/navigationHome"
```
Comment (hungld-1302) [inline app/src/main/res/menu/menu_bottom_navigation.xml]: navigation_home

File: app/src/main/java/com/ndt/marvelapp/ui/MainActivity.kt [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+class MainActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityMainBinding
+    override fun onCreate(savedInstanceState: Bundle?) {
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        super.onCreate(savedInstanceState)
+        requestWindowFeature(Window.FEATURE_CUSTOM_TITLE)
+        setTheme(R.style.Theme_MarvelApp)
+        setContentView(binding.root)
+        val navHostFragment =
+            supportFragmentManager.findFragmentById(R.id.fragment_nav_host) as NavHostFragment
```
Comment (hungvv-1301) [inline app/src/main/java/com/ndt/marvelapp/ui/MainActivity.kt]:
```suggestion
            findNavController()
```

File: app/build.gradle [hunk 8]
```diff
@@ -27,6 +29,7 @@ android {
 
     buildFeatures {
         viewBinding true
+        dataBinding true
```
Comment (daolq-2712) [inline app/build.gradle]: remove!
project 2 này mình chỉ dùng viewBinding e nhé!

File: app/build.gradle [hunk 9]
```diff
@@ -48,4 +51,34 @@ dependencies {
     androidTestImplementation TestLibraries.junitExt
     androidTestImplementation TestLibraries.junitExt
 
+    //rxjava3
+    implementation Libraries.rxjava3
```
Comment (daolq-2712) [inline app/build.gradle]: remove,
mình sẽ dùng courcoutine ở prj 2 e nhé!

File: app/build.gradle [hunk 10]
```diff
@@ -48,4 +51,34 @@ dependencies {
...
+    implementation Libraries.navigationUI
+    implementation Libraries.navigationDynamic
+
+    //roomdb
+    implementation Libraries.roomRuntime
+    implementation Libraries.roomRxjava3
+    implementation Libraries.roomKtx
+    kapt Libraries.roomCompiler
+
+    //thu vien load anh
```
Comment (daolq-2712) [inline app/build.gradle]:
```suggestion
    //load image 
```

File: app/src/main/AndroidManifest.xml [hunk 11]
```diff
@@ -10,11 +10,15 @@
         android:label="@string/app_name"
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
+        android:usesCleartextTraffic="true"
```
Comment (daolq-2712) [inline app/src/main/AndroidManifest.xml]: tại sao lại thêm line này vào e nhỉ?

File: app/src/main/java/com/ndt/marvelapp/base/BaseFragment.kt [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+        super.onViewCreated(view, savedInstanceState)
+        viewModel.error.observe(viewLifecycleOwner, Observer {
+            view.context.showToast(it)
+        })
+        setupViews()
+        initData()
+        initActions()
+    }
+
+    protected abstract fun setupViews()
```
Comment (daolq-2712) [inline app/src/main/java/com/ndt/marvelapp/base/BaseFragment.kt]: remove!

File: app/src/main/java/com/ndt/marvelapp/base/BaseViewModel.kt [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+open class BaseViewModel : ViewModel() {
+    private val disposables = CompositeDisposable()
+    private val _error = MutableLiveData<String>()
+    val error: LiveData<String>
+        get() = _error
+
+    override fun onCleared() {
+        super.onCleared()
+        disposables.clear()
```
Comment (daolq-2712) [inline app/src/main/java/com/ndt/marvelapp/base/BaseViewModel.kt]:
```suggestion
        disposables.clear()
        super.onCleared()
```

File: app/src/main/res/layout/fragment_home.xml (hunk 5 above)
Comment (daolq-2712) [reply app/src/main/res/layout/fragment_home.xml]: Vẫn chưa thêm blank line ở cuối file

**Review by daolq-2712 — CHANGES_REQUESTED** [review]
Pls check my comments!

**Pull request merged**